stopcovid/drill_progress/status.py

```diff
@@ -14,15 +20,23 @@ def handle_dialog_event_batches(batches: List[DialogEventBatch]):
             initiator.trigger_first_drill(batch.phone_number, str(batch.batch_id))
 
     user_repo = DrillProgressRepository()
+    state_repo = DynamoDBDialogRepository()
     for batch in batches:
         user_repo.update_user(batch)
-        if initiates_subsequent_drill(batch):
+        if initiates_subsequent_drill(batch, state_repo):
             initiator.trigger_next_drill_for_user(batch.phone_number, str(batch.batch_id))
 
 
 def initiates_first_drill(batch: DialogEventBatch):
     return any(event for event in batch.events if isinstance(event, UserValidated))
 
 
-def initiates_subsequent_drill(batch: DialogEventBatch):
-    return any(event for event in batch.events if isinstance(event, NextDrillRequested))
+def initiates_subsequent_drill(batch: DialogEventBatch, repo: DialogRepository):
+    for event in batch.events:
+        if isinstance(event, NextDrillRequested):
+            return True
+        elif isinstance(event, DrillCompleted):
+            dialog_state = repo.fetch_dialog_state(batch.phone_number)
```

Comment:
This will do a `n` queries where `n` is the number of events in the batch. Let's call fetch_dialog_state once for the entire batch, and pass the dialog_stage into this method

Reply:
Yeah, good catch. I moved it there thinking I could avoid calling it sometimes, but forgot it's looping through events.